userspace/libscap/scap_procs.c

```diff
@@ -188,15 +188,32 @@ int32_t scap_check_suppressed(struct scap_suppress* suppress, scap_evt *pevent,
 
 	default:
 
-		HASH_FIND_INT64(suppress->m_suppressed_tids, &(pevent->tid), stid);
+		if(suppress->m_suppressed_tids)
+		{
+			uint16_t slot = (cpuid & (SCAP_CACHE_CPUID_MAX - 1));
+			if(suppress->m_cpuid_tid_stid_cache[slot].tid == pevent->tid)
+			{
+				stid = suppress->m_cpuid_tid_stid_cache[cpuid].stid; // use cached
+			}
+			else
+			{
+				HASH_FIND_INT64(suppress->m_suppressed_tids, &(pevent->tid), stid);
+				suppress->m_cpuid_tid_stid_cache[slot].tid = pevent->tid; //  re-cache
+				suppress->m_cpuid_tid_stid_cache[slot].stid = stid; // re-cache
+			}
+		}
+		else
+		{
+			// suppressed TIDs hash table is not allocated / used
+			stid = NULL;
+		}
```

Comment:
@Andreagit97 the reason I consciously didn't change that particular business logic is because of the other complicated `if()` statement directly below the one you highlighted. It seemed like there was a larger change of introducing a bug in the logic by arbitrarily changing and merging the four (in total) `if(){}else{}` statements. So I was trying to be as safe as possible and laser focused just on the cache aspect... and "let sleeping dogs lie" so-to-speak :-)